response.go

```diff
@@ -7,17 +7,26 @@ import (
 	"github.com/vmihailenco/msgpack/v5"
 )
 
-type Response struct {
-	RequestId uint32
-	Code      uint32
-	// Error contains an error message.
-	Error string
-	// Data contains deserialized data for untyped requests.
-	Data []interface{}
-	// Pos contains a position descriptor of last selected tuple.
-	Pos      []byte
-	MetaData []ColumnMetaData
-	SQLInfo  SQLInfo
+type Response interface {
+	Decode() ([]interface{}, error)
+	DecodeTyped(res interface{}) error
+
+	Pos() []byte
+	MetaData() []ColumnMetaData
+	SQLInfo() SQLInfo
+}
+
+type ConnResponse struct {
+	requestId uint32
+	code      uint32
+	// error contains an error message.
+	error string
```

Comment:
Let's try to remove this fields (discussed in zoom).
```suggestion
	// requestId uint32
	code      uint32
	// error contains an error message.
	// error string
```